tests/core/test_parent_child.py

```diff
@@ -264,3 +269,59 @@ def discover_streams(self):
     # Parent record is emitted
     assert messages[10]
     assert messages[10]["type"] == SingerMessageType.RECORD
+
+
+def test_abstract_parent_streams(snapshot: Snapshot, snapshot_dir: Path):
+    """Test that abstract parent streams are skipped."""
+
+    class AbstractParent(Stream):
+        """An abstract parent stream."""
+
+        __abstract__ = True
+
+        def get_records(self, context):  # noqa: ARG002
+            yield {"id": 1}
+            yield {"id": 2}
+
+        def generate_child_contexts(self, record, context):  # noqa: ARG002
+            yield {"pid": record["id"]}
```

Comment:
Wondering if there is a better way to support this... It would be good if this could be encapsulated into `generate_child_contexts` only, but that currently requires at least one record (hence `get_records` implementation).